### PR TITLE
Removing kafka consuming repsonsibility out of the websocket client

### PR DESCRIPTION
**Pull request by Jason-RH**
The work dispatcher will handle consuming job messages from the job router. Also removed the go routines for write and dispatch out of read and into the handler to further simplify read.

### CODE REVIEW

File: internal/controller/dispatcher.go [hunk 1]
```diff
@@ -93,34 +95,68 @@ func (rd *ResponseDispatcher) Dispatch(ctx context.Context, m protocol.Message,
...
+	readerConfig *queue.ConsumerConfig
 }
 
 func (wd *WorkDispatcher) GetKey() string {
 	return fmt.Sprintf("%s:%s", wd.account, wd.nodeID)
 }
 
-func (wd *WorkDispatcher) Dispatch(ctx context.Context) {
-	fmt.Println("This is when the work dispatcher would consume a job from the jobs kafka topic")
+func (wd *WorkDispatcher) Dispatch(ctx context.Context, c chan<- Work) {
```
Comment (dehort) [inline internal/controller/dispatcher.go]: Let's think about renaming this method to more closely match what its doing.  Maybe something along the lines of StartDispatchingMessages?  I'm open to ideas.  I think of dispatch is a singular event and this is more of a continuous process.  What do you think?

File: internal/controller/dispatcher.go [hunk 2]
```diff
@@ -93,34 +95,68 @@ func (rd *ResponseDispatcher) Dispatch(ctx context.Context, m protocol.Message,
...
+	r := queue.StartConsumer(wd.readerConfig)
```
Comment (dehort) [inline internal/controller/dispatcher.go]: What do you think about having the factory pass a reference to the consumer in?  If we inject a consumer object into this code, then we can mock it out easier for testing.

File: internal/controller/dispatcher.go [hunk 3]
```diff
@@ -93,34 +95,68 @@ func (rd *ResponseDispatcher) Dispatch(ctx context.Context, m protocol.Message,
...
+		log.Printf("Kafka job reader - received message from %s-%d [%d]: %s: %s\n",
+			m.Topic,
+			m.Partition,
+			m.Offset,
+			string(m.Key),
+			string(m.Value))
+
+		if string(m.Key) == wd.GetKey() {
+			// FIXME:
+			w := Work{}
```
Comment (dehort) [inline internal/controller/dispatcher.go]: At this point, i think we're going to need to unmarshall what we get from kafka to convert it back into a Work struct.  

I think the code should look something like this:

```suggestion
var w controller.Work
err := json.Unmarshal(m.Value, &w)
if err != nil {
  log.Println("Unable to unmarshal work from kafka queue")
  continue
}		
```

File: internal/controller/dispatcher.go (hunk 1 above)
Comment (Jason-RH) [reply internal/controller/dispatcher.go]: Yep a name change would be useful here.

File: internal/controller/dispatcher.go (hunk 2 above)
Comment (Jason-RH) [reply internal/controller/dispatcher.go]: Added the consumer creation into the factory method. This should still allow for easier mocking when it comes to testing the dispatching method.

**Comment by Jason-RH** [comment]
@dehort The initialization of the message dispatcher is commented out until the job router is ready to use it.

File: internal/controller/ws/client.go [hunk 4]
```diff
@@ -27,6 +26,8 @@ type rcClient struct {
 
 	responseDispatcher *controller.ResponseDispatcher
 
+	messageDispatcher *controller.MessageDispatcher
```
Comment (dehort) [inline internal/controller/ws/client.go]: Just thinking a bit here...  In this case, I don't think the client/peer needs to keep up with the message dispatcher object.  You don't have to make any changes in this PR.  If we think its not really needed, we can remove it another PR.

Comment (Jason-RH) [reply internal/controller/ws/client.go]: Will add in a follow up PR

**Pull request merged**